### PR TITLE
Add tags to nuget version validation

### PR DESCRIPTION
**Pull request by matthewfcarlson**
This will allow -beta -rc and -alpha to nuget versions
Details can be found here: https://docs.microsoft.com/en-us/nuget/create-packages/prerelease-packages

### CODE REVIEW

File: edk2toolext/nuget_publishing.py [hunk 1]
```diff
@@ -207,15 +207,20 @@ def _MakeNuspecXml(self, ContentDir, ReleaseNotesText=None):
     def _GetNuPkgFileName(self, version):
         # Nuget removes leading zeros so to match we must do the same
         s = self.Name + "."
+        append_tag = None
         parts = version.split(".")
-        for a in parts:
-            s += str(int(a)) + "."
+        if "-" in parts[-1]:
+            parts[-1], append_tag = parts[-1].split("-")
```
Comment (corthon) [inline edk2toolext/nuget_publishing.py]: If you're using this multiple assignment here...

File: edk2toolext/environment/extdeptypes/nuget_dependency.py [hunk 2]
```diff
@@ -57,20 +57,31 @@ def GetNugetCmd():
     def normalize_version(version):
         if len(version) == 0:
             raise ValueError("Unparsable version: empty string")
-        version_parts = tuple(int(num) for num in version.split('.'))
-        if len(version_parts) > 4:
+        version_parts = version.split("-")
+        if len(version_parts) > 2:
+            raise ValueError("Unparsable version: '%s'!")
+        version_ints = version_parts[0]
```
Comment (corthon) [inline edk2toolext/environment/extdeptypes/nuget_dependency.py]: ... why not here?

File: edk2toolext/tests/test_nuget_dependency.py [hunk 3]
```diff
@@ -125,10 +125,18 @@ def test_normalize_version(self):
...
+        self.assertEqual(proper_version4, NugetDependency.normalize_version(version4))
         # try some bad cases
-        version4 = "not a number"
         with self.assertRaises(ValueError):
-            NugetDependency.normalize_version(version4)
+            NugetDependency.normalize_version("not a number")
+        with self.assertRaises(ValueError):
+            NugetDependency.normalize_version("6.0-beta-beta")
+        with self.assertRaises(ValueError):
+            NugetDependency.normalize_version("6.0-bad")
```
Comment (corthon) [inline edk2toolext/tests/test_nuget_dependency.py]: Can we add a test case for "3.2.1.-beta"?
Dash immediately after .
How about ---?

Comment (matthewfcarlson) [reply edk2toolext/tests/test_nuget_dependency.py]: Done

File: edk2toolext/environment/extdeptypes/nuget_dependency.py (hunk 2 above)
Comment (matthewfcarlson) [reply edk2toolext/environment/extdeptypes/nuget_dependency.py]: Fair

File: edk2toolext/nuget_publishing.py (hunk 1 above)
Comment (matthewfcarlson) [reply edk2toolext/nuget_publishing.py]: Fair

**Comment by matthewfcarlson** [comment]
@corthon Alright I think this is ready to go

**Comment by matthewfcarlson** [comment]
Also add tags to nuget publish https://docs.microsoft.com/en-us/nuget/reference/nuspec

**Comment by matthewfcarlson** [comment]
@spbrogan @corthon This should be ready

**Pull request merged**